tracer/src/Datadog.Trace/Logging/Internal/DatadogLoggingFactory.cs

```diff
@@ -163,13 +164,19 @@ private static string GetDefaultLogDirectory(string? logDirectory)
 #if NETFRAMEWORK
             logDirectory = Path.Combine(Environment.GetFolderPath(Environment.SpecialFolder.CommonApplicationData), @"Datadog .NET Tracer", "logs");
 #else
-            if (System.Runtime.InteropServices.RuntimeInformation.IsOSPlatform(System.Runtime.InteropServices.OSPlatform.Windows))
+            var isWindows = System.Runtime.InteropServices.RuntimeInformation.IsOSPlatform(System.Runtime.InteropServices.OSPlatform.Windows);
+
+            if (Serverless.GetIsAzureFunction())
```

Comment:
using `Serverless.IsAzureFunction` instead (which is initialized `usingGetIsAzureFunction()` always evaluates to false, but calling the function directly works